packages/loopring_v3/contracts/impl/owners/DelayedOwner.sol

```diff
@@ -0,0 +1,211 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../../lib/AddressUtil.sol";
+import "../../lib/BytesUtil.sol";
+import "../../lib/Claimable.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/ReentrancyGuard.sol";
+
+
+/// @title DelayedOwner
+/// @author Brecht Devos - <brecht@loopring.org>
+/// @dev Base class for an Owner contract where certain functions have
+///      a mandatory delay for security purposes.
+contract DelayedOwner is Claimable, ReentrancyGuard
+{
+    using AddressUtil for address payable;
+    using BytesUtil   for bytes;
+    using MathUint    for uint;
+
+    struct Transaction
+    {
+        uint timestamp;
+        uint value;
+        bytes data;
+    }
+
+    event TransactionDelayed(
+        uint    delayedTransactionIdx,
+        uint    timestamp,
+        uint    value,
+        bytes   data,
+        uint    delay
+    );
+
+    event TransactionExecuted(
+        uint    timestamp,
+        uint    value,
+        bytes   data
+    );
+
+    event DelayedTransactionExecuted(
+        uint    delayedTransactionIdx,
+        uint    timestamp,
+        uint    value,
+        bytes   data
+    );
+
+    address public contractAddress;
+    mapping (bytes4 => uint) public functionDelays;
```

Comment:
I think we should make the  delay configurations ar queryable, something like `delayedFunctions()`

Reply:
Done